src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -35,6 +35,34 @@
 
 ;; FUNCTIONS
 
+(defn deregister-unused-drivers
+  "Given a database config or connection pool, deregister the driver
+   (hsql or postgres) that isn't being used, and return the initial driver
+   list for reregistration in case it's required."
+  [db-config]
+  (let [configured-driver (java.sql.DriverManager/getDriver
+                            (if-let [datasource (:datasource db-config)]
+                              (.getJdbcUrl datasource)
+                              (format "jdbc:%s:%s"
+                                      (:subprotocol db-config)
+                                      (:subname db-config))))
```

Comment:
I'd be tempted to abstract this to a function on `db-config`, `(defn get-driver [db-config])` or something like that.